lib/rubocop/config_loader.rb

```diff
@@ -58,7 +58,11 @@ def merge(base_hash, derived_hash)
 
       def base_configs(path, inherit_from)
         configs = Array(inherit_from).map do |f|
-          f = File.join(File.dirname(path), f) unless f.start_with?('/')
+          if f.include?('~')
+            f = File.expand_path(f)
+          else
+            f = File.join(File.dirname(path), f) unless f.start_with?('/')
+          end
```

Comment:
Oh, maybe passing `path` as a second argument to `File.expand_path` will make it work? From [the docs](http://ruby-doc.org/core-2.2.0/File.html#method-c-expand_path):

> Relative paths are referenced from the current working directory of the process unless `dir_string` is given, in which case it will be used as the starting point.

It appears this is how `path` is used in the else branch.